VideoLocker/src/main/java/org/edx/mobile/view/DownloadListActivity.java

```diff
@@ -56,68 +38,57 @@ public void onItemClicked(DownloadEntry model) {
             }
 
             @Override
-            public void onDownloadComplete(DownloadEntry model) {
-                //showDownloadCompleteView();
-                adapter.remove(model);
-                adapter.notifyDataSetChanged();
-            }
-
-            @Override
             public void onDeleteClicked(DownloadEntry model) {
-                if(environment.getStorage().removeDownload(model) >= 1){
+                if (environment.getStorage().removeDownload(model) >= 1) {
                     // update the list data as one download is removed
-                    try{
-                        adapter.remove(model);
-                        adapter.notifyDataSetChanged();
-                    }catch(Exception e){
-                        logger.error(e);
-                    }
-
+                    adapter.remove(model);
+                    adapter.notifyDataSetChanged();
                 }
             }
         };
-
-        adapter.setStore(environment.getStorage());
+        final ListView downloadListView = (ListView) findViewById(R.id.my_downloads_list);
         downloadListView.setAdapter(adapter);
-        final ArrayList<DownloadEntry> list = new ArrayList<DownloadEntry>();
+    }
+
+    private void observeOngoingDownloads() {
         environment.getDatabase().getListOfOngoingDownloads(new DataCallback<List<VideoModel>>() {
             @Override
-            public void onResult(List<VideoModel> result) {
-                if(result!=null){
-                    for(VideoModel de : result){
-                        list.add((DownloadEntry) de);
-                    }
-                    adapter.setItems(list);
+            public void onResult(final List<VideoModel> result) {
+                if (result != null) {
+                    handler.post(new Runnable() {
```

Comment:
Hmm, it looks like the `DataCallback` base implementation already has support for having the result delivered in the UI thread using a `Handler`, but for some reason this is not enabled in the default constructor. I suggest that we change the default, as I imagine this could be the source of other bugs as well (and from what I remember most of the callback methods seem to be designed to interact with the UI in any case.